mongo/errors.go

```diff
@@ -592,6 +595,40 @@ func (bwe BulkWriteException) HasErrorCodeWithMessage(code int, message string)
 // serverError implements the ServerError interface.
 func (bwe BulkWriteException) serverError() {}
 
+// ServerSelectionError is a server selection error encountered running any command
+// besides commitTransaction in a transaction.
+type ServerSelectionError struct {
```

Comment:
Why was this solution chosen instead of an existing pattern such as "IsTimeout"? For example

```go
func IsServerSelectionError(err error) {	
	for ; err != nil; err = unwrap(err) {
		if _, ok := err.(topology.ServerSelectionError); ok {
			return true
		}
	}

	return false
}
```

Reply:
We need to return an error with a label "TransientTransactionError" for the server selection error if I understand the [spec](https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#transienttransactionerror) correctly. Also, returning a wrapper from the "mongo" package is better than exposing a type in the "x" package in my opinion.

A helper function to check the SeverSelectionError may be useful but I don't think necessary with the label already attached.

Reply:
@qingyang-hu I have a concerns about wrapping x package errors at the mongo-package level, specifically that we have not done this in the past. This solution still exposes the x package error and extends the overall error tree. We also have an existing pattern for handling x package error assertions such as when we check for "mongo.IsTimeout". In the context of this ticket, we could use a more general assertion function "IsTransientTransactionError()" which only checks for errors labeled with TransientTransactionError:

```go
func IsTransientTransactionError(err error) {
		if le, ok := err.(labeledError); ok {
			if le.HasErrorLabel(drivers.TransientTransactionError) {
				return true
			}
		}
}
```

This (1) minimizes the API's surface area, (2) isn't specific to a particular error tree, (3) and could avoid addition references to the "x" package by replacing "drivers.TransientTransactionError" with a local constant or `"TransientTransaction:error"`.

Reply:
AFAIK there are two primary use cases for adding the "TransientTransactionError" label:
1. Users can check for the "TransientTransactionError" label on errors.
2. `WithTransaction` can check for the "TransientTransactionError" label [here](https://github.com/mongodb/mongo-go-driver/blob/14777c12f1e412190bd65750b46b1de155e3f5eb/mongo/session.go#L208) and [here](https://github.com/mongodb/mongo-go-driver/blob/14777c12f1e412190bd65750b46b1de155e3f5eb/mongo/session.go#L253).

It seems like this change supports case 2, but it's not clear what the most common way for users to accomplish case 1 is yet. We should investigate how users currently check for error labels before deciding to add new helper functions.